test/testcne/tailqs_test.c

```diff
@@ -0,0 +1,157 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ * Copyright (c) <2019-2022>, Intel Corporation. All rights reserved.
+ */
+
+#include <stdio.h>             // for NULL, snprintf, EOF
+#include <string.h>            // for strcmp, strncmp
+#include <getopt.h>            // for getopt_long, option
+#include <cne_common.h>        // for CNE_SET_USED, __cne_unused
+#include <tst_info.h>          // for TST_ASSERT_EQUAL_AND_CLEANUP, tst_end, TST_A...
+#include <pthread.h>
+#include <cne_tailq.h>
+
+#include "tailqs_test.h"
+
+#define do_return(...)                                        \
+    do {                                                      \
+        printf("Error at %s, line %d: ", __func__, __LINE__); \
+        printf(__VA_ARGS__);                                  \
+        return 1;                                             \
+    } while (0)
+
+static struct cne_tailq_elem tst_dummy_tailq = {
+    .name = "dummy",
+};
+CNE_REGISTER_TAILQ(tst_dummy_tailq)
+
+static struct cne_tailq_elem tst_dummy_dyn_tailq = {
+    .name = "dummy_dyn",
+};
+static struct cne_tailq_elem tst_dummy_dyn2_tailq = {
+    .name = "dummy_dyn",
+};
+
+static struct cne_tailq_entry d_elem;
+static struct cne_tailq_entry d_dyn_elem;
+
+static int
+test_tailq_early(void)
+{
+    struct cne_tailq_entry_head *d_head;
+
+    d_head = CNE_TAILQ_CAST(tst_dummy_tailq.head, cne_tailq_entry_head);
+    if (d_head == NULL)
+        do_return("Error %s has not been initialised\n", tst_dummy_tailq.name);
+
+    /* check we can add an item to it */
+    TAILQ_INSERT_TAIL(d_head, &d_elem, next);
+
+    return 0;
+}
+
+static int
+test_tailq_create(void)
+{
+    struct cne_tailq_entry_head *d_head;
+
+    /* create a tailq and check its non-null (since we are post-eal init) */
+    if ((cne_tailq_register(&tst_dummy_dyn_tailq) < 0) || (tst_dummy_dyn_tailq.head == NULL))
+        do_return("Error allocating %s\n", tst_dummy_dyn_tailq.name);
+
+    d_head = CNE_TAILQ_CAST(tst_dummy_dyn_tailq.head, cne_tailq_entry_head);
+
+    /* check we can add an item to it */
+    TAILQ_INSERT_TAIL(d_head, &d_dyn_elem, next);
+
+    if (strcmp(tst_dummy_dyn2_tailq.name, tst_dummy_dyn_tailq.name))
+        do_return("Error, something is wrong in the tailq test\n");
```

Comment:
This error message is not helpful. Woudl be better to describe what's wrong iwth the test,.